project-base/tests/App/Acceptance/acceptance/PageObject/Front/ProductDetailPage.php

```diff
@@ -17,8 +17,9 @@ class ProductDetailPage extends AbstractPage
      */
     public function addProductIntoCart($quantity = 1)
     {
-        $this->tester->fillFieldByCss(
-            self::PRODUCT_DETAIL_QUANTITY_INPUT,
+        $productQuantityInput = $this->tester->findElementByCss(self::PRODUCT_DETAIL_QUANTITY_INPUT);
+        $this->tester->fillFieldByElement(
+            $productQuantityInput,
```

Comment:
🤔 can you please explain to me, why locate element by css first and then fill by the element, instead of fill by css?

At first look, it's harder to read it for me and I don't see added value, so I need some enlightenment here :)

Reply:
Ok, added value is that methods with elements do scroll to element as user would, also it is then visible on screen if some problem occurs. Another thing it does it that it moves mouse cursor over it, because in some designs on product lists for example some buttons and inputs might appear after user moves cursor over product there. This happens on commerce cloud.

But! You made me think and I agree that it should be unnecessary, so I have updated StrictWebDriver way, that all methods there use WebDriverElements - see 222a245f3c3d1cd5dc4f49aa3b52bb9d1bfba7aa 5274c7071d3c70739683b22a5fe85245ac08ba61 eae6477978d7ec020aa891bc8f13848d0e3234c1 d2fb74ddb32023526478c0b5354709cc84ca9933

Reply:
🆒🆒